ctapipe/image/waveform_reducer.py

```diff
@@ -147,21 +165,20 @@ def extract_charge(self, waveforms):
         """
 
 
-class FullIntegrator(ChargeExtractor):
+class FullWaveformSum(WaveformReducer):
     """
-    Charge extractor that integrates the entire waveform.
+    Waveform reducer that integrates the entire waveform.
     """
 
-    def extract_charge(self, waveforms):
-        # TODO: remove integration window return
-        peakpos = np.zeros(waveforms.shape[:2], dtype=np.intp)
-        window = np.ones(waveforms.shape, dtype=np.bool)
-        return waveforms.sum(2), peakpos, window
+    def __call__(self, waveforms):
+        charge = waveforms.sum(2)
+        pulse_time = extract_pulse_time_weighted_average(waveforms)
+        return charge, pulse_time
 
 
-class SimpleIntegrator(ChargeExtractor):
+class UserWindowSum(WaveformReducer):
```

Comment:
I would remove the `User` here.

Reply:
I disagree, all of these extracts do a window sum, but for this one the window is specified manually by the user

Reply:
So ... `GlobalPeakWindowSum` needs to be called `UserGlobalPeakWindowSum` because the user has to manually the width and the shift?

Reply:
Not really...the user does not manually select a fixed window position in that case, it is calculated from the waveform

Reply:
okay fine.

Reply:
I strongly oppose calling anything `User`. 
Name it like what it needs or does. `StaticWindowSum` or similar.

User does not tell anything and most people will not share the idea that a user of ctapipe should be concerned with integration schemas for raw data.